R/bundle.R

```diff
@@ -164,7 +164,12 @@ bundleApp <- function(appName, appDir, appFiles, appPrimaryDoc, assetTypeName,
                       contentCategory) {
 
   # infer the mode of the application from its layout
-  appMode <- inferAppMode(appDir, appPrimaryDoc, appFiles)
+  # unless we're an API, in which case, we're API mode.
+  if (contentCategory == "api") {
```

Comment:
Prefer comparing strings using `identical` -- using `==` results in errors if one of the operands is NULL, e.g. 

    > if (NULL == "api") { }
    Error in if (NULL == "api") { : argument is of length zero